src/runner/browser-env/vite/browser-modules/driver.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    WebDriverProtocol,
+    MJsonWProtocol,
+    JsonWProtocol,
+    AppiumProtocol,
+    ChromiumProtocol,
+    SauceLabsProtocol,
+    SeleniumProtocol,
+    GeckoProtocol,
+    WebDriverBidiProtocol,
+} from "@wdio/protocols";
+import { webdriverMonad, sessionEnvironmentDetector, devtoolsEnvironmentDetector } from "@wdio/utils";
+import { getEnvironmentVars } from "webdriver";
+import { SOCKET_MAX_TIMEOUT, SOCKET_TIMED_OUT_ERROR, MAX_ARGS_LENGTH } from "./constants.js";
+import { BrowserEventNames } from "./types.js";
+
+type ProtocolCommandFn = (...args: unknown[]) => Promise<unknown>;
+type PropertiesObject = Record<string, PropertyDescriptor>;
+
+const COMMANDS_HANDLED_BY_SERVER = ["debug", "saveScreenshot", "savePDF"];
+
+export default class ProxyDriver {
+    static newSession(
+        params: Record<string, unknown>,
+        modifier: never,
+        userPrototype: Record<string, PropertyDescriptor>,
+        commandWrapper: VoidFunction | undefined,
+    ): unknown {
+        const monad = webdriverMonad(params, modifier, getWdioPrototype(userPrototype));
+        return monad(window.__testplane__.sessionId, commandWrapper);
+    }
+}
+
+function getWdioPrototype(userPrototype: Record<string, PropertyDescriptor>): PropertiesObject {
+    for (const commandName of COMMANDS_HANDLED_BY_SERVER) {
+        delete userPrototype[commandName];
+    }
+
+    const prototype = {
+        ...getMockedProtocolCommands(),
+        ...getEnvironmentFlags(),
+        ...userPrototype,
+    };
+
+    return prototype;
+}
+
+function getAllProtocolCommands(): string[] {
+    const protocols = [
+        WebDriverProtocol,
+        MJsonWProtocol,
+        JsonWProtocol,
+        AppiumProtocol,
+        ChromiumProtocol,
+        SauceLabsProtocol,
+        SeleniumProtocol,
+        GeckoProtocol,
+        WebDriverBidiProtocol,
+    ];
+    const commands = new Set<string>();
+
+    for (const protocol of protocols) {
+        for (const endpointData of Object.values(protocol)) {
+            for (const { command } of Object.values<{ command: string }>(endpointData)) {
+                commands.add(command);
+            }
+        }
+    }
+
+    return Array.from(commands);
+}
+
+function getMockedProtocolCommands(): PropertiesObject {
+    return [...getAllProtocolCommands(), ...COMMANDS_HANDLED_BY_SERVER, ...window.__testplane__.customCommands].reduce(
```

Comment:
mock all custom commands by default

Reply:
Would it work with chainings like `browser.$('.element').click()`?

I noticed you are not splitting `customCommands` for browser and elements, and just using them together, so if a user adds a custom command to the element, would he be able to use it,

Reply:
> Would it work with chainings like browser.$('.element').click()?

Yes, because I mock only protocol commands and do nothing with wrappers like `browser.$` and others. Implementation of command `$` - https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/commands/browser/%24.ts. And protocol commands implement inside `findElement` helper - https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/commands/browser/%24.ts#L108. And at the end it returns element instanse on which another commands can be called.

> I noticed you are not splitting customCommands for browser and elements, and just using them together, so if a user adds a custom command to the element, would he be able to use it like browser.$('.element').customCommand?

It gives an error from the server that command is not exists on element. Llike this - `Can't call "custom_command" on element with selector "body", it is not a function`